example/poisson/poisson_driver.cpp

```diff
@@ -74,28 +146,93 @@ TaskCollection PoissonDriver::MakeTaskCollection(BlockList_t &blocks) {
+    auto update = solver.AddTask(mat_elem, poisson_package::UpdatePhi<MeshData<Real>>,
                                  md.get(), mdelta.get());
 
-    auto send =
-        solver.AddTask(update, parthenon::cell_centered_bvars::SendBoundaryBuffers, md);
-
-    auto recv = solver.AddTask(
-        start_recv, parthenon::cell_centered_bvars::ReceiveBoundaryBuffers, md);
-
-    auto setb =
-        solver.AddTask(recv | update, parthenon::cell_centered_bvars::SetBoundaries, md);
-
-    auto clear = solver.AddTask(send | setb, &MeshData<Real>::ClearBoundary, md.get(),
-                                BoundaryCommSubset::all);
+    auto norm = solver.AddTask(update, poisson_package::SumDeltaPhi<MeshData<Real>>,
+                               mdelta.get(), &update_norm.val);
+    solver_region.AddRegionalDependencies(4, i, norm);
+    auto start_reduce_norm =
+        solver.AddTask(norm, &AllReduce<Real>::StartReduce, &update_norm, MPI_SUM);
+    auto finish_reduce_norm =
+        solver.AddTask(start_reduce_norm, &AllReduce<Real>::CheckReduce, &update_norm);
+    auto report_norm = (i == 0 ? solver.AddTask(
+                                     finish_reduce_norm,
+                                     [](Real *norm) {
+                                       if (Globals::my_rank == 0) {
+                                         std::cout << "Update norm = " << *norm
+                                                   << std::endl;
+                                       }
+                                       *norm = 0.0;
+                                       return TaskStatus::complete;
+                                     },
+                                     &update_norm.val)
+                               : none);
+
+    auto send = solver.AddTask(update, cell_centered_bvars::SendBoundaryBuffers, md);
+
+    auto recv =
+        solver.AddTask(start_recv, cell_centered_bvars::ReceiveBoundaryBuffers, md);
+
+    auto setb = solver.AddTask(recv | update, cell_centered_bvars::SetBoundaries, md);
+
+    auto clear = solver.AddTask(send | setb | report_norm, &MeshData<Real>::ClearBoundary,
+                                md.get(), BoundaryCommSubset::all);
 
     auto check = solver.SetCompletionTask(
         clear, poisson_package::CheckConvergence<MeshData<Real>>, md.get(), mdelta.get());
     // mark task so that dependent tasks (below) won't execute
     // until all task lists have completed it
-    solver_region.AddRegionalDependencies(0, i, check);
-
-    auto print = tl.AddTask(check, poisson_package::PrintComplete);
+    solver_region.AddRegionalDependencies(5, i, check);
+
+    auto print = none;
+    if (i == 0) { // only print once
+      print = tl.AddTask(check, poisson_package::PrintComplete);
+    }
+    //--- End of tasks related to solving the Poisson eq
+
+    // do a vector reduction (everything below here), just for fun
+    // first fill it in
```

Comment:
Looks good, I'll be using this in AthenaPK soon

Reply:
Great, let me know if you run into issues.